pkg/agent/agent.go

```diff
@@ -838,32 +835,32 @@ func (i *Initializer) initNodeLocalConfig() error {
 			i.patchNodeAnnotations(nodeName, types.NodeTransportAddressAnnotationKey, nil)
 		}
 	}
-	i.networkConfig.TransportIface = transportInterfaceName
 
 	// Update the Node's MAC address in the annotations of the Node. The MAC address will be used for direct routing by
 	// OVS in noencap case on Windows Nodes. As a mixture of Linux and Windows nodes is possible, Linux Nodes' MAC
 	// addresses should be reported too to make them discoverable for Windows Nodes.
 	if i.networkConfig.TrafficEncapMode.SupportsNoEncap() {
 		klog.Infof("Updating Node MAC annotation")
-		if err := i.patchNodeAnnotations(nodeName, types.NodeMACAddressAnnotationKey, localIntf.HardwareAddr.String()); err != nil {
+		if err := i.patchNodeAnnotations(nodeName, types.NodeMACAddressAnnotationKey, transportInterface.HardwareAddr.String()); err != nil {
 			return err
 		}
 	}
 
 	i.nodeConfig = &config.NodeConfig{
-		Name:                  nodeName,
-		OVSBridge:             i.ovsBridge,
-		DefaultTunName:        defaultTunInterfaceName,
-		NodeIPv4Addr:          nodeIPv4Addr,
-		NodeIPv6Addr:          nodeIPv6Addr,
-		NodeTransportIPv4Addr: transportIPv4Addr,
-		NodeTransportIPv6Addr: transportIPv6Addr,
-		UplinkNetConfig:       new(config.AdapterNetConfig),
-		NodeLocalInterfaceMTU: localIntf.MTU,
-		WireGuardConfig:       i.wireGuardConfig,
-	}
-
-	mtu, err := i.getNodeMTU(localIntf)
+		Name:                       nodeName,
+		OVSBridge:                  i.ovsBridge,
+		DefaultTunName:             defaultTunInterfaceName,
+		NodeIPv4Addr:               nodeIPv4Addr,
+		NodeIPv6Addr:               nodeIPv6Addr,
+		NodeTransportInterfaceName: transportInterface.Name,
+		NodeTransportIPv4Addr:      transportIPv4Addr,
+		NodeTransportIPv6Addr:      transportIPv6Addr,
+		UplinkNetConfig:            new(config.AdapterNetConfig),
+		NodeTransportInterfaceMTU:  transportInterface.MTU,
+		WireGuardConfig:            i.wireGuardConfig,
+	}
+
+	mtu, err := i.getNodeMTU(transportInterface)
```

Comment:
I noticed that the parameter name for `getNodeMTU` is still `localIntf`. Maybe it could be renamed to `transportInterface` for consistency.

Reply:
done